.github/workflows/ci.yml

```diff
@@ -0,0 +1,38 @@
+name: CI
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    continue-on-error: ${{ matrix.experimental }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: [2.7, jruby, jruby-head, truffleruby-head]
+        experimental: [false]
+        include:
+          - ruby: head
+            experimental: true
+          - ruby: truffleruby
+            experimental: true
+
+    env:
+      JAVA_OPTS: '-Xmx1024m'
+      RUBYOPT: '-w'
+      JRUBY_OPTS: '--dev'
+
+    name: "Tests: Ruby ${{ matrix.ruby }}"
+    steps:
+    - name: Clone Repo
+      uses: actions/checkout@v2
+    - name: Setup system Ruby ${{ matrix.ruby }}
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: ${{ matrix.ruby }}
+        bundler-cache: true # runs 'bundle install' and caches installed gems automatically
+    - name: Run tests
+      run: |
+        gem install bundler --version 1.17.3
```

Comment:
I don’t know that `gem install bundler` achieves anything really (other than making CI slower). Per the `ruby/setup-ruby` [docs](https://github.com/ruby/setup-ruby#caching-bundle-install-automatically):

>Note that any step doing `bundle install` (for the root `Gemfile`) or `gem install bundler` can be removed with `bundler-cache: true`.

At this point the gems have already been installed so it doesn’t really matter what version of Bundler we then install.

Reply:
Yes it's probably good to copy what Travis did first and then simplify and refine.

Reply:
Perhaps more helpfully: the right way to control the Bundler version is to [provide a `bundler:` input](https://github.com/ruby/setup-ruby/blob/master/action.yml#L12-L18) to the `ruby/setup-ruby` action.

Reply:
Indeed, it's counter-productive to install bundler here, and will probably be more confusing than helpful.
Also Bundler 1 is basically not really supported on Ruby 2.7+ (no longer maintained AFAIK).

Reply:
=> https://github.com/ruby-concurrency/concurrent-ruby/pull/916